astropy/io/fits/hdu/compressed.py

```diff
@@ -666,6 +668,24 @@ def __init__(self, data=None, header=None, name=None,
         self._orig_bscale = self._bscale
         self._orig_bitpix = self._bitpix
 
+    @property
+    def name(self):
+        if 'EXTNAME' not in self._header:
+            return self._default_name
+
+        name = self._default_name
+
+        # If there are multiple EXTNAME values use the first that
+        # is not COMPRESSED_IMAGE
+        indices = self._header._keyword_indices['EXTNAME']
```

Comment:
I'm not entirely sure if I'm allowed to use the `_keyword_indices` method but I couldn't find a public API that would let me find which cards to test.